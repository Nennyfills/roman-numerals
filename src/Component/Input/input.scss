.input {
  input,
  label {
    display: block;
    font-weight: 500;
    font-size: 16px;
    line-height: 8px;
    padding-bottom: 8px;
    color: #222222;
    text-align: left;
  }
  input {
    width: 100%;
    display: inline-block;
    text-transform: uppercase;
    margin: 0.6rem 0.5rem 0.8rem 0;
    border-radius: 5px;
    border: 1px solid #dfe2e6;
    font-style: normal;
    padding: 12px;
    font-weight: normal;
    text-align: left;
    &::-webkit-outer-spin-button,
    &::-webkit-inner-spin-button {
      -webkit-appearance: none;
    }
    &[type='number'] {
      -moz-appearance: textfield;
    }

    &:focus {
      outline: none;
    }
  }
}
