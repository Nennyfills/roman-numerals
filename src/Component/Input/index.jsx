import React from 'react';
import PropTypes from 'prop-types';

import './input.scss';

const Input = ({
  type,
  label,
  placeholder,
  handleChange,
  name,
  value,
  inputId,
}) => (
  <div className="input">
    <label htmlFor={inputId}>{label}</label>
    <input
      data-testid={inputId}
      id={inputId}
      name={name}
      value={value}
      type={type}
      placeholder={placeholder}
      onChange={handleChange}
      autoComplete="off"
    />
  </div>
);

Input.defaultProps = {
  label: '',
  placeholder: '',
  name: '',
  value: '',
  handleChange: () => 'handleChange',

  inputId: 'test',
};
Input.propTypes = {
  type: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  handleChange: PropTypes.func,
  name: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  inputId: PropTypes.string,
};
export default Input;
