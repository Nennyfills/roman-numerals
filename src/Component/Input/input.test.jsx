import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Input from './index';

describe('Test Input component', () => {
  test('renders a message', () => {
    const { container } = render(<Input type="text" />);
    expect(container.firstChild).toMatchSnapshot(`
      <div />
    `);
  });
  test('Test Input text', () => {
    const { getByTestId } = render(<Input type="text" />);
    const input = getByTestId('test');

    fireEvent.change(input, { target: { value: 'xxx' } });
  });
  test('Test Input number', () => {
    const { getByTestId } = render(<Input type="number" />);
    const input = getByTestId('test');

    fireEvent.change(input, { target: { value: 98 } });
  });
});
