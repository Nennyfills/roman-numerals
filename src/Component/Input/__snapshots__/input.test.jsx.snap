// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input component renders a message: 
      <div />
     1`] = `
<div
  class="input"
>
  <label
    for="test"
  />
  <input
    autocomplete="off"
    data-testid="test"
    id="test"
    name=""
    placeholder=""
    type="text"
    value=""
  />
</div>
`;
